rewindable-index/test/RewindableIndex/Spec/Sqlite.hs

```diff
@@ -0,0 +1,289 @@
+module RewindableIndex.Spec.Sqlite where
+
+import Control.Lens.Operators ((^.))
+import Control.Lens.TH qualified as Lens
+import Control.Monad (forM_)
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad.Trans.Maybe (MaybeT (MaybeT, runMaybeT))
+import Data.Foldable (foldl', toList)
+import Data.Functor ((<&>))
+import Database.SQLite.Simple qualified as Sql
+import Database.SQLite.Simple.FromField (FromField (fromField))
+import Database.SQLite.Simple.FromRow (FromRow (fromRow), field)
+import Database.SQLite.Simple.ToField (ToField (toField))
+import Database.SQLite.Simple.ToRow (ToRow (toRow))
+import GHC.Generics (Generic)
+import Safe (atMay)
+import Test.QuickCheck (Property)
+import Test.QuickCheck.Monadic (PropertyM, monadicIO)
+
+import RewindableIndex.Model (Conversion (Conversion, cHistory, cMonadic, cNotifications, cView), Index,
+                              IndexView (IndexView, ixDepth, ixSize, ixView), getFunction)
+import RewindableIndex.Model qualified as Ix
+import RewindableIndex.Storable (Buffered (getStoredEvents, persistToStorage), HasPoint (getPoint),
+                                 QueryInterval (QEverything, QInterval), Queryable (queryStorage),
+                                 Resumable (resumeFromStorage), Rewindable (rewindStorage), State, StorableEvent,
+                                 StorablePoint, StorableQuery, StorableResult, config, emptyState,
+                                 filterWithQueryInterval, memoryBufferSize)
+import RewindableIndex.Storable qualified as Storable
+
+newtype Handle = Handle Sql.Connection
+
+data instance StorablePoint Handle =
+    Point Int
+  | Genesis
+  deriving stock (Eq, Show, Generic)
```

Comment:
`stock` is default here and not required yes? I think it is good to put it in and in that case we should add it to `StorableEvent`, few lines below, as well

Reply:
There used to be a lot more deriving statements there and I forgot to remove the stock when I removed those. Good catch!